### PR TITLE
Raw reader: use posix_spawn instead of fork + exec

### PR DESCRIPTION
**Pull request by 0xxon**
This commit switchexisd the Raw reader to use posix_spawn, instead of the combination of fork + exec. This should be much more efficient, and also makes the code smaller, and easier to read and understand.

The github diff looks much worse than this is - after line 187 this is essentially only indentation changes.

### CODE REVIEW

**Review by awelzel — COMMENTED** [review]
One comment about memory management.

File: src/input/readers/raw/Raw.cc [hunk 1]
```diff
@@ -136,141 +139,113 @@ bool Raw::Execute()
 		return false;
 		}
 
-	childpid = fork();
-	if ( childpid < 0 )
+	posix_spawnattr_t attrs;
+	posix_spawnattr_init(&attrs);
```
[the remark this answers is not in the document]
Comment (0xxon) [reply src/input/readers/raw/Raw.cc:147]: Hi,

I noticed that a lot of these operations can fail with ENOMEN while writing. I kind of decided to go the way that a lot of Zeek goes so far - which is to assume that "if we are out of memory, things are going south and Zeek will crash in any case".

If you think that makes no sense, I am happy to change this.

I (think) I fixed the memory issues - thanks a lot.

**Review by awelzel — APPROVED** [review]

File: src/input/readers/raw/Raw.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (0xxon) [reply src/input/readers/raw/Raw.cc]: Ok, I just added error handling to everything - could you take a look if this seems fine?

[the remark this answers is not in the document]
Comment (awelzel) [reply src/input/readers/raw/Raw.cc]: I had approved to let you make the call but realized my comment didn't mention that. Newest changes look fine to me.

You're obviously right that if we're actually short on memory that `reporter->Error()` is likely the next thing to blow :-)

**Review by awelzel — APPROVED** [review]
Assuming CI passes.

**Pull request merged**